spring-integration-webflux/src/test/java/org/springframework/integration/webflux/dsl/WebFluxDslTests.java

```diff
@@ -145,6 +163,82 @@ public void setup() {
 						.build();
 	}
 
+	@Test
+	public void testTimeoutForWebFluxOutboundGateway() {
+		ClientHttpConnector httpConnector = new HttpHandlerConnector((request, response) -> {
+			response.setStatusCode(HttpStatus.OK);
+			response.getHeaders().setContentType(MediaType.TEXT_PLAIN);
+
+
+			DataBufferFactory bufferFactory = response.bufferFactory();
+			return response.writeWith(Mono.just(bufferFactory.wrap("FOO\nBAR\n".getBytes()))
+					.delayElement(Duration.ofMillis(11000)))
+					.then(Mono.defer(response::setComplete));
+		});
+
+		WebClient webClient = WebClient.builder()
+				.clientConnector(httpConnector)
+				.build();
+
+		new DirectFieldAccessor(this.webFluxFlowWithTimeout)
+				.setPropertyValue("webClient", webClient);
+
+		QueueChannel replyChannel = new QueueChannel();
+
+		Message<String> testMessage =
+				MessageBuilder.withPayload("test")
+						.setReplyChannel(replyChannel)
```

Comment:
I think we shouldn't use a `setReplyChannel` for this kind of test. Since you use the same instance for both reply and error responses, I was confused first about exception assertion in the end. I was expected from "error channel", but I saw a `replyChannel`.  So, your test-case mislead me to investigate how do we process exceptions like that timeout 😢